### DISCUSSION TITLE
A tax system could be good.

### DISCUSSION THREAD

**Discussion by Mcdikmen** [Zrips/CMI#5163] [Feature Requests]
Hello,

I want to suggest u a tax system. It's simple and very effective for some server types i think. Cuz i dont want my players reaching too many balances. If i am tax my player balances every week like a percentage (and offline players too), it could be good. Maybe some players may receive more taxes. Like:

0-10.000 K balances, recieving %5
10.000-50.000 K balances, recieving %10
50.000+ balances, recieving %20

Hope you like it. Thanks for reading.

**Comment by th3lucas** [comment]
Hello!

I hope i am not late to the party. :)
I would love a CMI Tax system too - would be a nice benefit to a economy plugin!
Players get a lot of money quickly (which is a great feeling for players - but it can destroy a healthy economy.) 

As @Mcdikmen said:

- Let Admins decide which tax rate should be applied for a balance.
- Taxes should be collected for offline players too.

**Comment by AlmostPanda** [comment]
Any update yet?

**Comment by CloudeLecaw** [comment]
you can create a cmi chedule that triggeres
once a week and takes money from all offline players and all online players (you need to make it with two commands, one for all online and one for all offline)

```
cmi money [pay/give/take/set] [playerName/all/alloffline] [amount/%rand/1-1000%/1%[min-max][playerName]] (-s)
```

```
cmi money take all 5% -s

and

cmi money take alloffline 5% -s
```

also you could set limitation to the tax. here the example fron the wiki.

/cmi money take Zrips 1%[100-500][Zhax] takes from Zrips balance 1%, but no less then 100 and no more then 500 of what Zhax has

the only limitation here that a schedule has is that your limited to a fixed tax for all.

**Reply by AlmostPanda** [reply]
is there also an option to send the collected taxes from all players and all offline players on a player or offline player balance?

**Reply by CloudeLecaw** [reply]
well for us on our server its possible because we can create and modfy placeholders that are global and not player related.

if you cab do this too, in this case its possible.
- set (add) 5% money of from all players online to placeholder X
- set (add) 100% money of bankofflineplayer to placeholder X
- take 100% of money from  bankofflineplayer
- set (add) 5% of money from all offline players to placeholder X
- take 5% of money of all player online
- take 5% of money of all players offline
- send money with amount of placeholder X to bankofflineplayer
- reset placeholder X

if you cant create and modfy own public placeholder its not really possible because the bankofflineplayer also would pay 5% of what he has currently. This would make the money he should havent at the end to an wrong balance amount.

so at least - no its not possible with cmi alone.
